src/facebook-social-provider.js

```diff
@@ -67,21 +67,40 @@ FacebookSocialProvider.prototype.getOAuthTokenInteractive_ =
  * Loads contacts of the logged in user, and calls this.addUserProfile_
  * and this.updateUserProfile_ (if needed later, e.g. for async image
  * fetching) for each contact.
+ * This method is called recursively to load each page of contacts.
  */
-FacebookSocialProvider.prototype.loadContacts_ = function() {
-  this.facebookGet_('me/friends').then(function(resp) {
-    var users = resp.data;
+FacebookSocialProvider.prototype.loadContacts_ = function(url) {
+  if (!this.loginState_) {
+    throw 'Error in FacebookSocialProvider.loadContacts_: not logged in';
+  }
+
+  var LIMIT = 100;
+  if (!url) {
+    url = 'https://graph.facebook.com/v2.1/me/friends' +
+      '?access_token=' + this.loginState_.authData.facebook.accessToken +
+      '&format=json&redirect=false&limit=' + LIMIT;
+  }
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  var thisFacebookSocialProvider = this;
```

Comment:
thisFacebookSocialProvider is used within the xhr.onload callback.  Within that callback, just using ```this``` won't give us what we want, and if we do the usual ```.bind(this)``` we won't be able to easily access this.response returned by the XHR